File: 1653. Minimum Deletions to Make String Balanced/minimumDeletions.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumDeletions = function (s) {
    let b_cnt = 0
    let ans = 0

    for (let chr of s) {
        if (chr === 'b') b_cnt++
        else if (b_cnt > 0) {
            b_cnt--
            ans++
        }
    }
    return ans
};

File: 1653. Minimum Deletions to Make String Balanced/minimumDeletions.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int minimumDeletions(string s) {
        int b_cnt= 0;
        int ans = 0;

        for(auto& chr : s){
            if(chr == 'b')b_cnt++;
            else if(b_cnt > 0){
                b_cnt--;
                ans++;
            }
        }
        return ans;
    }
};